moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -237,9 +299,16 @@ class RobotTrajectory
    * \param end_index - index of last traj point of the part to append from the source traj, the default is to add until
    * the end of the source traj
    */
-  RobotTrajectory& append(const RobotTrajectory& source, double dt, size_t start_index = 0,
+  RobotTrajectory& append(const RobotTrajectory& source, rclcpp::Duration dt, size_t start_index = 0,
                           size_t end_index = std::numeric_limits<std::size_t>::max());
 
+  [[deprecated("Use of double for duration is deprecated, please pass in rclcpp::Duration")]] RobotTrajectory&
+  append(const RobotTrajectory& source, double dt, size_t start_index = 0,
+         size_t end_index = std::numeric_limits<std::size_t>::max())
+  {
+    return append(source, rclcpp::Duration::from_seconds(dt), start_index, end_index);
+  }
```

Comment:
Why is there a mix between some functions being defined in this header and some functions merely being declared in it?

Reply:
That is a problem that already exists here and deserves its own cleanup.